templates/deployment.yaml

```diff
@@ -117,12 +117,19 @@ spec:
         securityContext:
           {{- toYaml .Values.securityContext | nindent 10 }}
         volumeMounts:
-        - name: worker-config
+        - name: worker-shared-volume
           readOnly: true
           mountPath: /etc/config
+        {{- if .Values.existingSecret }}
```

Comment:
I think this condition can be removed otherwise the secret provided from the chart wouldn't get mounted.